x/upgrade/test/integration_test.go

```diff
@@ -0,0 +1,202 @@
+		accounts[i] = tmrand.Str(9)
+	}
+	blockTime := time.Millisecond * 300
+	tmCfg := config.DefaultConfig()
+	tmCfg.Consensus.TimeoutCommit = blockTime
+	tmCfg.RPC.ListenAddress = fmt.Sprintf("tcp://127.0.0.1:%d", getFreePort())
+	tmCfg.P2P.ListenAddress = fmt.Sprintf("tcp://127.0.0.1:%d", getFreePort())
+	tmCfg.RPC.GRPCListenAddress = fmt.Sprintf("tcp://127.0.0.1:%d", getFreePort())
+
+	// change the tally params in the genesis so we can pass votes quickly
+	s.ecfg = encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	gdgs := v1.DefaultGenesisState()
+	gdgs.TallyParams.Quorum = "0.000001"
+	gdgs.TallyParams.Threshold = "0.000001"
+
+	genState, kr, err := testnode.DefaultGenesisState(accounts...)
+	require.NoError(t, err)
+
+	// replace the default genesis state with the modified one
+	genState[gov.AppModuleBasic{}.Name()] = s.ecfg.Codec.MustMarshalJSON(gdgs)
+
+	tmNode, capp, cctx, err := testnode.New(t, testnode.DefaultParams(), tmCfg, false, genState, kr, tmrand.Str(6))
+	require.NoError(t, err)
+
+	cctx, stopNode, err := testnode.StartNode(tmNode, cctx)
+	require.NoError(t, err)
+
+	appConf := testnode.DefaultAppConfig()
+	appConf.GRPC.Address = fmt.Sprintf("127.0.0.1:%d", getFreePort())
+	appConf.API.Address = fmt.Sprintf("tcp://127.0.0.1:%d", getFreePort())
+
+	cctx, cleanupGRPC, err := testnode.StartGRPCServer(capp, appConf, cctx)
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		t.Log("tearing down testnode")
+		require.NoError(t, stopNode())
+		require.NoError(t, cleanupGRPC())
+	})
+	s.accounts = accounts
+	s.cctx = cctx
+	require.NoError(t, s.cctx.WaitForNextBlock())
+
+	// set the gov module address
+	// retrieve the gov module account via grpc
+	aqc := authtypes.NewQueryClient(s.cctx.GRPCClient)
+	resp, err := aqc.ModuleAccountByName(
+		s.cctx.GoContext(), &authtypes.QueryModuleAccountByNameRequest{Name: "gov"},
+	)
+	s.Require().NoError(err)
+	var acc authtypes.AccountI
+	err = s.ecfg.InterfaceRegistry.UnpackAny(resp.Account, &acc)
+	s.Require().NoError(err)
+
+	s.govModuleAddress = acc.GetAddress().String()
+}
+
+func (s *UpgradeTestSuite) unusedAccount() string {
+	s.mut.Lock()
+	acc := s.accounts[s.accountCounter]
+	s.accountCounter++
+	s.mut.Unlock()
+	return acc
+}
+
+func (s *UpgradeTestSuite) TestLegacyGovUpgradeFailure() {
+	t := s.T()
+
+	dep := sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewInt(1000000000000)))
+	acc := s.unusedAccount()
+	accAddr := getAddress(acc, s.cctx.Keyring)
+
+	sftwr := types.NewSoftwareUpgradeProposal("v1", "Social Consensus", types.Plan{
+		Name:   "v1",
+		Height: 20,
+		Info:   "rough social consensus",
+	})
+
+	msg, err := v1beta1.NewMsgSubmitProposal(sftwr, dep, accAddr)
+	require.NoError(t, err)
+
+	// submit the transaction and wait a block for it to be included
+	res, err := testnode.SignAndBroadcastTx(s.ecfg, s.cctx.Context, acc, msg)
+	require.NoError(t, err)
+
+	// require.Equal(t, abci.CodeTypeOK, res.Code) // we're only submitting the tx, so we expect everything to work
```

Comment:
Why is this commented? Shouldn't this also pass?

Reply:
good catch! https://github.com/celestiaorg/celestia-app/pull/1629/commits/62f6a5fb1c579139c81970363dc8e48f8508e893

I think it was just a casuality of the debug process